packages/ui/src/bounty/hooks/useBounties.ts

```diff
@@ -1,24 +1,80 @@
+import { useMemo } from 'react'
+
 import { useGetBountiesCountQuery, useGetBountiesQuery } from '@/bounty/queries'
-import { BountyOrderByInput } from '@/common/api/queries'
+import { BountyOrderByInput, BountyStage, BountyWhereInput } from '@/common/api/queries'
 import { usePagination } from '@/common/hooks/usePagination'
 import { SortOrder, toQueryOrderByInput } from '@/common/hooks/useSort'
 
+import { BountyFiltersState } from '../components/BountiesFilters'
 import { asBounty } from '../types/casts'
 
-interface UseBountiesProps {
+type BountyStatus = 'active' | 'past'
+
+export interface UseBountiesProps {
   order: SortOrder<BountyOrderByInput>
   perPage?: number
+  filters?: BountyFiltersState
+  status: BountyStatus
 }
 
-export const useBounties = ({ order, perPage = 10 }: UseBountiesProps) => {
+export const useBounties = ({ order, perPage = 10, filters, status }: UseBountiesProps) => {
   const orderBy = toQueryOrderByInput<BountyOrderByInput>(order)
   const { data: dataCount } = useGetBountiesCountQuery()
   const totalCountPerPage = dataCount?.bountiesConnection.totalCount
   const { offset, pagination } = usePagination(perPage, totalCountPerPage ?? 0, [order])
 
-  const { loading, data } = useGetBountiesQuery({
-    variables: { orderBy: orderBy ?? [], limit: perPage, offset: offset },
-  })
+  const variables = useMemo(() => {
+    const where: BountyWhereInput = {}
+
+    if (filters?.search) {
+      where.title_contains = filters.search
+    }
+
+    if (status === 'past') {
+      where.stage_eq = BountyStage['Terminated']
+    } else {
+      where.stage_in = [
+        BountyStage['Funding'],
+        BountyStage['WorkSubmission'],
+        BountyStage['Judgment'],
+        BountyStage['Expired'],
+        BountyStage['Successful'],
+        BountyStage['Failed'],
+      ]
+    }
```

Comment:
Looks kind of dumb, but there's no such field as _stage_ne_, so I can't see a prettier solution